app/src/main/webapp/js/services.js

```diff
@@ -42,51 +42,84 @@ esbMessageAdminServices.service('EsbMessageService', ['$http', '$q', 'Globals',
     };
 
     self.getSyncKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/Entities"});
+        return self.getTree("Entities");
     };
 
     self.getSearchKeysTree = function() {
-        return $http({method: 'GET', url: "api/key/tree/SearchKeys"});
+        return self.getTree("SearchKeys");
     };
 
-    self.addKey = function(argParentId, argName, argType, argValue) {
+    self.respondSuccess = function(argResponse) {
+        if(argResponse.status==="Success") {
+            return argResponse.tree;
+        } else {
+            messageCenterService.add('danger', argResponse.errorMessage, {status: messageCenterService.status.permanent});
+            return null;
+        }
+    }
+
+    self.respondError= function()  {
+        messageCenterService.add('danger', 'Unable to contact server!', {status: messageCenterService.status.permanent});
+        return null;
+    }
+
+    self.getTree = function(argType) {
+        return $http.get("api/key/tree/"+argType).
+        success(function(response){
+            self.respondSuccess(response);
+        }).
+        error(function(){
+            self.respondError();
+        });
+    };
 
-        return $http({method: 'PUT', url: "api/key/addChild/{parentId}?name={name}&type={type}&value={value}".
-            supplant({parentId: argParentId, name: argName, type: argType, value: argValue })}).
-            then(function(results) {
+    self.addKey = function(argParentId, argName, argType, argValue) {
 
-                return results;
+        return $http.put("api/key/addChild/{parentId}?name={name}&type={type}&value={value}".
+            supplant({parentId: argParentId, name: argName, type: argType, value: argValue })).
+            success(function(response) {
+                self.respondSuccess(response);
+            }).
+            error(function(){
+                self.respondError();
             });
-            // todo error handling
     };
 
     self.updateKey = function(argId, argName, argType, argValue) {
 
-        return $http({method: 'PUT', url: "api/key/update/{id}?name={name}&type={type}&value={value}".
-            supplant({id: argId, name: argName, type: argType, value: argValue })}).
-            then(function(results) {
-
-                return results;
+        return $http.put("api/key/update/{id}?name={name}&type={type}&value={value}".
+            supplant({id: argId, name: argName, type: argType, value: argValue })).
+            success(function(response) {
+                self.respondSuccess(response);
+            }).
+            error(function(){
+                self.respondError();
             });
-            // todo error handling
     };
 
     self.deleteKey = function(argId) {
 
-        return $http({method: 'PUT', url: "api/key/delete/{id}".
-            supplant({id: argId })}).
-            then(function(results) {
-
-                return results;
+        return $http.put("api/key/delete/{id}".
```

Comment:
It's interesting that your API has verbs in the url but all of the HTTP methods use PUT/POST. Why not follow semantics of GET/PUT/POST/DELETE all with /api/key/{id} ?

I'm not suggesting changing this so much as wondering the background for that.